src/components/services-view/services-view.controller.ts

```diff
@@ -93,14 +128,55 @@ export class ServicesViewController implements angular.IController {
     this.$scope.$emit('open-overlay-panel', item);
   };
 
+  private filterForKeywords (searchText: string, items: any) {
+    let keywords = this.keywordService.generateKeywords(searchText);
+    return this.keywordService.filterForKeywords(items, ['name', 'tags'], keywords);
+  }
+
+  private filterChange = (filters: any) => {
+
+    this.filterByCategory(this.ctrl.currentFilter, this.ctrl.currentSubFilter, false);
+
+    this.ctrl.filterConfig.appliedFilters = filters;
+
+    if (filters && filters.length > 0) {
+      _.each(filters, (filter) => {
+        this.ctrl.filteredItems = this.filterForKeywords(filter.value, this.ctrl.filteredItems);
+      });
+    }
+
+    this.updateFilterControls();
+  };
+
+  private clearAppliedFilters() {
+    this.ctrl.filterConfig.appliedFilters = [];
+  }
+
   private resizeExpansion() {
     $('.services-sub-category').removeAttr('style');
     let activeCat = $('.services-sub-category.active');
-    let contentHeight = activeCat.find('.services-items').innerHeight();
+    let contentHeight = activeCat.find('.services-items-container').innerHeight() - 20;
+    contentHeight += activeCat.find('.services-items').innerHeight();
     activeCat.css('margin-bottom', contentHeight + 'px');
   }
 
   private updateActiveCardStyles() {
     this.$timeout(this.resizeExpansion, 50);
   }
+
+  private updateFilterControls() {
+    this.$timeout(() => {
+      if (this.ctrl.filterConfig.appliedFilters.length > 0) {
+        this.ctrl.filterConfig.resultsCount = this.ctrl.filteredItems.length;
+        $('.toolbar-pf-results h5').text(this.ctrl.filterConfig.resultsCount + ' of ' + this.ctrl.totalCount + ' items');
+        $('.toolbar-pf-results p').first().text('Active Filters:');
+      } else {
+        $('.toolbar-pf-results h5').text(this.ctrl.totalCount + ' items');
+
+        if (this.ctrl.totalCount <= 1) {
+          $('.filter-pf.filter-fields input').attr('disabled', '');
```

Comment:
This should be available via Angular Patternfly as well if this is a desired behavior.